mne/viz/tests/test_ica.py

```diff
@@ -163,6 +163,12 @@ def test_plot_ica_sources():
                            ecg=False, eog=False, exclude='bads')
     ica = ICA(n_components=2, max_pca_components=3, n_pca_components=3)
     ica.fit(raw, picks=ica_picks)
+    ica.exclude = [1]
+    fig = ica.plot_sources(raw)
+    fig.canvas.key_press_event('escape')
+    # Sadly close_event isn't called on Agg backend and the test always passes.
+    assert_array_equal(ica.exclude, [1])
```

Comment:
I couldn't find a reasonable way to test the interactive selection of components on 'Agg' backend, but this should catch the error on other backends.